src/app/hooks/useViewTracker/index.jsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState, useContext } from 'react';
+import { useInView } from 'react-intersection-observer';
+// Polyfill IntersectionObsesrver, e.g. for IE11
+import 'intersection-observer';
+
+import { ServiceContext } from '#contexts/ServiceContext';
+import { RequestContext } from '#contexts/RequestContext';
+import { buildATIClickParams } from '#containers/ATIAnalytics/params';
+import { getComponentInfo } from '#app/lib/analyticsUtils/index';
+import { sendEventBeacon } from '#containers/ATIAnalytics/beacon';
+
+const VIEWED_DURATION_MS = 1000;
+
+const useViewTracker = ({ pageData, componentName, actionLabel }) => {
+  const requestContext = useContext(RequestContext);
+  const serviceContext = useContext(ServiceContext);
+  const [viewSent, setViewSent] = useState(false);
+  const [ref, inView] = useInView({
+    threshold: 0.5,
+  });
+  const eventTrackingProps = buildATIClickParams(
+    pageData,
+    requestContext,
+    serviceContext,
+  );
+
+  useEffect(() => {
+    let timeout;
+    const componentInfo = getComponentInfo({
+      result: window.location.href,
+      componentName,
+      componentData: {
+        actionLabel,
+        child: 'link', // TODO ask Jon B what this should be
+      },
+    });
+
+    if (inView) {
+      timeout = setTimeout(() => {
+        if (!viewSent) {
+          setViewSent(true);
+          sendEventBeacon({
+            type: 'view',
+            componentName,
+            service: serviceContext.service,
+            componentInfo,
+            ...eventTrackingProps,
+          });
```

Comment:
A slight observation, these events are 'fire and forget' if they fail we have no retry logic. I'd say that's correct as this is just tracking data. 

It's good case for clientside monitoring as we have the potential for misleading stats if we encounter bugs sending the beacon in certain scenarios: https://github.com/bbc/simorgh-infrastructure/issues/1347